CHANGELOG.md

```diff
@@ -1,5 +1,53 @@
+* Altogether, this allows you to track conversion events and attribute them to variations even when you don't know all of a user's attribute values, and even if the user's attribute values or the experiment's configuration have changed such that the user is no longer affected by the experiment. As a result, **you may observe an increase in the conversion rate for previously-instrumented events.** If that is undesirable, you can reset the results of previously-running experiments after upgrading to the 3.0 SDK.
+* This will also allow you to attribute events to variations from other Optimizely projects in your account, even though those experiments don't appear in the same datafile.
+* Note that for results segmentation in Optimizely results, the user attribute values from one event are automatically applied to all other events in the same session, as long as the events in question were actually received by our backend. This behavior was already in place and is not affected by the 3.0 release.
+* Support for all types of attribute values, not just strings.
+* All values are passed through to notification listeners.
+* Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³]. 
+* Strings, booleans, and valid numbers are relevant for audience conditions.
+* Support for additional matchers in audience conditions:
+* An `exists` matcher that passes if the user has a non-null value for the targeted user attribute and fails otherwise.
+* A `substring` matcher that resolves if the user has a string value for the targeted attribute.
+* `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
+* The original (`exact`) matcher can now be used to target booleans and valid numbers, not just strings.
+* Support for A/B tests, feature tests, and feature rollouts whose audiences are combined using `"and"` and `"not"` operators, not just the `"or"` operator.
+* Datafile-version compatibility check: The SDK will remain uninitialized (i.e., will gracefully fail to activate experiments and features) if given a datafile version greater than 4.
+* Updated Pull Request template and commit message guidelines.
+* Only remove old variation assignments from the user profile service when there are >= 100 persisted assignments.
+* Support for empty bucketing IDs when evaluating feature rollouts.
+
+### Breaking Changes:
+* Previously, notification listeners were only given string-valued user attributes because only strings could be passed into various method calls. That is no longer the case. The ActivateListener and TrackListener interfaces now receive user attributes as `NSDictionary<NSString *, NSObject *> * _Nullable` instead of  `NSDictionary<NSString *,NSString *> * _Nonnull`.
```

Comment:
```suggestion
* Previously, notification listeners were only given string-valued user attributes because only strings could be passed into various method calls. That is no longer the case. The `ActivateListener` and `TrackListener` interfaces now receive user attributes as `NSDictionary<NSString *, NSObject *> * _Nullable` instead of  `NSDictionary<NSString *,NSString *> * _Nonnull`.
```
Sorry, I think I updated the central doc after you'd first loaded it!

Reply:
@jaeopt @thomaszurkan-optimizely is this the correct terminology? Is it appropriate to call these things "interfaces" that "receive" user attributes? I didn't really follow this part of the code.

Reply:
Yay, @jaeopt says this is okay!